compiler/rustc_privacy/src/lib.rs

```diff
@@ -1207,6 +1215,30 @@ impl<'tcx> Visitor<'tcx> for TypePrivacyVisitor<'tcx> {
         intravisit::walk_ty(self, hir_ty);
     }
 
+    fn visit_infer(&mut self, inf: &'tcx hir::InferArg) {
+        self.span = inf.span;
+        if let Some(typeck_results) = self.maybe_typeck_results {
+            if let Some(ty) = typeck_results.node_type_opt(inf.hir_id) {
+                if self.visit(ty).is_break() {
+                    return;
+                }
+            }
+        } else {
+            let local_id = self.tcx.hir().local_def_id(inf.hir_id);
+            if let Some(did) = self.tcx.opt_const_param_of(local_id) {
```

Comment:
for this to work you have to modify `fn opt_const_param_of` to also run for `GenericArg::Infer`, don't you?

For that we probably want to remove the `filter(|arg| arg.is_const())` and such and instead look at all const and type parameters to find the right one. Feel free to either add a `FIXME` in that method or implement it in this PR

Reply:
I changed them all to match on `GenericArg::Infer | GenericArg::Const` which may need to be changed since some code depends on matching on `GenericParamDefKind::Const`, but need to double check that